libbeat/publisher/processing/processing.go

```diff
@@ -38,3 +48,15 @@ type Supporter interface {
 	Create(cfg beat.ProcessingConfig, drop bool) (beat.Processor, error)
 	Close() error
 }
+
+// SetUnderAgent sets that the processing pipeline is being ran under the elastic-agent.
```

Comment:
I think we already have a way to do this by looking for the `management.enabled` flag: https://github.com/cmacknz/beats/blob/5337e63aa62646f8a62303d6de51c8ae723afa99/libbeat/common/fleetmode/fleet_mode.go#L26-L29

This lives in a separate package so it can be Apache licensed.

Reply:
Definitely just using an atomic bool would be faster if this is unconditionally executing on every event in the pipeline.

Reply:
I think using the atomic bool is much faster. So I think its best to keep this the way it is.